File: json/repeat/pub.js
var zmq = require('zmq')
, pub = zmq.socket('pub')
, command = {
    "cmd_header":{
        "receiver": "mbtcp",
        "sender":   "restful",        
        "version":  "1",
        "tid":      33,                
        "method":   "mbtcp.once.write"
    },
    "mb_tcp_header":{
        "ip": "192.168.1.1", 
        "port": 503,
        "id": 22
    },
    "requests": [ 
        {
            "code":     1,                 
            "register": 2003,             
            "value":    "1025",            
            "type":     "int64",        
            "alias":    "hello_1"  
        }, 
        {
            "code":     1,                 
            "register": 2003,             
            "value":    "1025",            
            "type":     "int64",        
            "alias":    "hello_1"  
        }
    ]
}

pub.connect("ipc:///tmp/dummy"); // connect to zmq endpoint

setInterval(function() {
    pub.send("mbtcp.once.write", zmq.ZMQ_SNDMORE);
    pub.send(JSON.stringify(command));
}, 1000); // emit every 1 seconds

File: json/repeat/pub.go
package main

import (
	"encoding/json"
	"fmt"
	zmq "github.com/taka-wang/zmq3"
	"time"
)

var (
	command = MbTcpMultipleWriteReq{ // named key
		CmdHeader: CmdHeader{
			Receiver: "core",
			Sender:   "me",
			Version:  "1",
			Tid:      32,
			Method:   "mbtcp.once.write",
		},
		MbTcpHeader: MbTcpHeader{
			Ip:   "192.168.1.1",
			Port: 503,
			Id:   22,
		},
		Requests: []MbWriteRequest{
			MbWriteRequest{
				Code:     1,
				Register: 2003,
				Value:    "1025",
				Type:     "int64",
				Alias:    "hello",
			},
			MbWriteRequest{
				Code:     1,
				Register: 2003,
				Value:    "1025",
				Type:     "int64",
				Alias:    "hello",
			},
		},
	}
)

func main() {
	pub()
}

func pub() {

	// marshal to json string
	cmd, err := json.Marshal(command)
	if err != nil {
		fmt.Println("json err:", err)
	}

	// zmq
	sender, _ := zmq.NewSocket(zmq.PUB)
	defer sender.Close()
	sender.Connect("ipc:///tmp/dummy")

	for {
		sender.Send("mbtcp.once.write", zmq.SNDMORE)
		sender.Send(string(cmd), 0)
		time.Sleep(time.Duration(1000) * time.Millisecond)
	}
}
